engines/libpmem.c

```diff
@@ -226,10 +211,11 @@ static int fio_libpmem_close_file(struct thread_data *td, struct fio_file *f)
 	struct fio_libpmem_data *fdd = FILE_ENG_DATA(f);
 	int ret = 0;
 
-	dprint(FD_IO,"DEBUG fio_libpmem_close_file\n");
-	dprint(FD_IO,"td->o.odirect %d \n",td->o.odirect);
+	dprint(FD_IO, "DEBUG fio_libpmem_close_file\n");
+	dprint(FD_IO, "td->o.odirect %d td->o.sync_io %d\n",
+					td->o.odirect, td->o.sync_io);
 
-	if (!td->o.odirect) {
+	if (!td->o.odirect || !td->o.sync_io) {
 		dprint(FD_IO,"pmem_drain\n");
 		pmem_drain();
```

Comment:
shouldn't we call here `pmem_persist`?

Reply:
No, the !sync_io condition means that the memcpy flag is NODRAIN.

I think the the direct flag should be removed from this condition - non-temporals and temporal both call a drain.

Reply:
yeah, right, direct should be removed from this equation at all.

We're talking about the close_file function, which IMHO should call drain/persist if it was not called previously (and this is the case if all of the IO was not sync).... or am I wrong and if user wants NO_DRAIN then NO_DRAIN it is, in the whole FIO workflow...?
Not that this is important at this point... maybe we should just remove this drain here...?

Reply:
I've removed the drain from the close function at all, seems not necessary.